marklogic-client-api/src/main/java/com/marklogic/client/datamovement/JacksonCSVSplitter.java

```diff
@@ -0,0 +1,94 @@
+package com.marklogic.client.datamovement;
+
+import java.io.InputStream;
+import java.io.Reader;
+import java.util.Iterator;
+import java.util.Spliterator;
+import java.util.Spliterators;
+import java.util.stream.Stream;
+import java.util.stream.StreamSupport;
+
+import com.fasterxml.jackson.databind.JsonNode;
+import com.fasterxml.jackson.databind.ObjectReader;
+import com.fasterxml.jackson.dataformat.csv.CsvMapper;
+import com.fasterxml.jackson.dataformat.csv.CsvParser;
+import com.fasterxml.jackson.dataformat.csv.CsvSchema;
+import com.marklogic.client.io.JacksonHandle;
+
+/**
+ * The JacksonCSVSplitter class uses the Jackson CSV parser without attempting to abstract it capabilities. 
+ * The application can override defaults by configuring the Jackson ObjectReader and CsvSchema including parsing TSV
+ */
+public class JacksonCSVSplitter implements Splitter<JacksonHandle> {
+    private CsvSchema csvSchema = null;
+    private CsvMapper csvMapper;
+    private ObjectReader objectParser;
+    private long count;
+    
+    public JacksonCSVSplitter() {
+        setCsvMapper();
+        setCount(0);
+    }
+    
+    public JacksonCSVSplitter(CsvSchema newSchema) {
+        setCsvSchema(newSchema);
+        setCsvMapper();
+        setCount(0);
+    }
+    
+    private void setCsvMapper() {
```

Comment:
On looking at the concrete refactoring, I'm concluding that the Technical Story has a design error.

We always want to convert CSV to JsonNodes, so we should expose ObjectReader.

Instead, we should expose CsvMapper.  That is, instead of having a setCsvMapper() that initializes a private field, we should have a private factory that constructs a default mapper only if the user didn't provide a CsvMapper.  

Then, in the configureObjReader() private factory, we can construct a private ObjectReader for JsonNode objects.